### ISSUE TITLE
API design: resources= or set_resources= ?

### ISSUE THREAD

**Issue by paugier** [snek5000/snek5000#186]
https://github.com/snek5000/snek5000/pull/185 deprecated 

```python
sim.make.exec('run_fg', resources={'nproc': 2})
```

in favor of

```python
sim.make.exec('run_fg', set_resources={'nproc': 2})
```

I don't really understand this change so I'm asking...

In terms of user interface, I think `set_` does not add a lot.

snakemake has both arguments

```
  --resources [NAME=INT ...], --res [NAME=INT ...]
                        Define additional resources that shall constrain the
                        scheduling analogously to --cores (see above). A
                        resource is defined as a name and an integer value.
                        E.g. --resources mem_mb=1000. Rules can use resources
                        by defining the resource keyword, e.g. resources:
                        mem_mb=600. If now two rules require 600 of the
                        resource 'mem_mb' they won't be run in parallel by the
                        scheduler. In cluster/cloud mode, this argument will
                        also constrain the amount of resources requested from
                        the server. (See https://snakemake.readthedocs.io/en/s
                        table/snakefiles/rules.html#resources-remote-execution
                        for more info) (default: None)

  --set-resources RULE:RESOURCE=VALUE [RULE:RESOURCE=VALUE ...]
                        Overwrite resource usage of rules. This allows to
                        fine-tune workflow resources. In particular, this is
                        helpful to target certain cluster nodes by e.g.
                        defining a certain partition for a rule, or overriding
                        a temporary directory. Thereby, VALUE has to be a
                        positive integer or a string, RULE has to be the name
                        of the rule, and RESOURCE has to be the name of the
                        resource. (default: None)
```

But anyway, do we need to exactly follow snakemake?

It seems that snakemake needs to be called as

```
snakemake -j1 --set-resources='run_fg:nproc=2' run_fg
# or
snakemake -j1 --resources=nproc=2 run_fg
```

which is really a strange notation but this is a bit another story...

**Comment by ashwinvis** [comment]
If you look at `doc/debug` directory. It seems like assigning `resources` does not do anything at all. I found it strange, since I recall `resources` working while we added this.

In the API, the counterpart of `--set-resources` was `overwrite_resources={'run_fg': {'nproc': 2}}`. I made it a bit simpler by defining `set_resources` since it was unused by Snakemake and it follows the CLI.

We dont have to follow it exactly, let me know if it is cumbersome and I can revert it.

**Comment by paugier** [comment]
Thanks, I understand better now. I still don't understand what happens with snakemake with `resources` but let's go with `set_resources`.

However, I think it would make sense to support also just `sim.make.exec('run_fg', nproc=2)` (because we use so many time `set_resources={'nproc': 2}`). I can make a PR for this if you think it is fine @ashwinvis.

**Comment by ashwinvis** [comment]
Yes sure, go ahead.
